packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js

```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
 			]);
 			expectUpdateEvent(childType, childCode, ['isChildOf']);
 		});
+
+		it('should not send extra UPDATE events when removing rich relationship property that was already non-existent', async () => {
+			const [main, child] = await Promise.all([
+				createMainNode(),
+				createNode(childType, { code: childCode }),
+			]);
+			await connectNodes(main, 'HAS_CURIOUS_CHILD', child);
+
+			const { status } = await patchHandler()(
+				getInput(
+					{
+						someString: 'some string',
+						curiousChild: {
+							code: childCode,
+							someString: null,
+						},
+					},
+					{ relationshipAction: 'replace', upsert: true },
+				),
+			);
+
+			expect(status).toBe(200);
+			expect(emitSpy).toHaveBeenCalledTimes(1);
```

Comment:
this is a jest spy which keeps an eye on the event emitter built into the handler function